src/main/webapp/app/shared/notification/notification-popup/notification-popup.component.ts

```diff
@@ -96,6 +116,24 @@ export class NotificationPopupComponent implements OnInit {
         }
     }
 
+    /**
+     * Adds a notification about a updated exercise during a live exam to the component's state
+     * and pushes updated problemStatement to student exam exercise via BehaviorSubjects
+     *
+     * @param notification {Notification}
+     */
+    private addExamUpdateNotification(notification: Notification): void {
+        if (notification.target) {
```

Comment:
Yes.
I can and have changed this.

The problem is that other parts of the notification system's legacy code have similar issues with a lot of nested code.
(and many more issues)
So I try to make my new code as "clean" as possible but will have to go back to the older code and refactor it in a later PR (otherwise this PR will get way too big I think).